gravitee-apim-console-webui/src/management/api/portal/ng-subscriptions/components/change-end-date-dialog/api-portal-subscription-change-end-date-dialog.component.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+
+import { PlanSecurityType } from '../../../../../../entities/management-api-v2';
+
+export interface ChangeEndDateData {
+  applicationName: string;
+  securityType: PlanSecurityType;
+  currentEndDate: Date;
+}
+export interface ChangeEndDateResult {
+  endDate: Date;
+}
+@Component({
+  selector: 'api-portal-subscription-change-end-date-dialog',
+  template: require('./api-portal-subscription-change-end-date-dialog.component.html'),
+  styles: [require('./api-portal-subscription-change-end-date-dialog.component.scss')],
+})
+export class ApiPortalSubscriptionChangeEndDateDialogComponent implements OnInit {
+  form: FormGroup;
+  data: ChangeEndDateData;
+  minDate: Date = new Date();
+
+  constructor(private readonly dialogRef: MatDialogRef<Date>, @Inject(MAT_DIALOG_DATA) dialogData: ChangeEndDateData) {
```

Comment:
```suggestion
  constructor(private readonly dialogRef: MatDialogRef<ApiPortalSubscriptionChangeEndDateDialogComponent, ChangeEndDateResult>, @Inject(MAT_DIALOG_DATA) dialogData: ChangeEndDateData) {
```

Attention to the type here.  Unfortunately, I've already let it go on other PRs.